src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/JudicialUserProfileProcessor.java

```diff
@@ -44,6 +57,25 @@ public void process(Exchange exchange) {
 
         audit(judicialUserProfileJsrValidatorInitializer, exchange);
 
+        //Get Elink ids from current load
+        validElinksInUserProfile = filteredJudicialUserProfiles.stream()
+            .map(userProfile -> userProfile.getElinksId()).collect(toSet());
+
+        //Get Elinks from previous loads
+        validElinksInUserProfile.addAll(loadElinksId());
+
+        filteredJudicialUserProfiles.stream()
+            .map(userProfile -> userProfile.getElinksId()).collect(toSet());
+
```

Comment:
why is it done this way? why cant we get the elinks id from db only?

Reply:
code at line 61 and 67 is repeated and can be avoided if u fetch it from db directly?